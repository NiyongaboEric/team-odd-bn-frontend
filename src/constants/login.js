const loginInputs = [
  { input: 'Email Address', className: 'email', type: 'email' },
  { input: 'Password', className: 'password', type: 'password' },
];

export default loginInputs;
