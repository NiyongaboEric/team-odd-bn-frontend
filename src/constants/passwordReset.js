export const emailBox = [
  {
    input: 'Email address', className: 'email', type: 'email', title: 'email should be valid',
  },
];
export const passwordBox = [
  {
    input: 'New password',
    className: 'password',
    type: 'password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    title:
             'A valid password should have a character, number, UPPER CASE letter and a lower case letter and should be longer than 8',
  },
  {
    input: 'Confirm password',
    className: 'confirmPassword',
    type: 'password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    title:
             'A valid password should have a character, number, UPPER CASE letter and a lower case letter and should be longer than 8',
  },
];
