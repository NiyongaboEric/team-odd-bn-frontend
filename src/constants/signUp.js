export const signupInputs = [
  {
    input: 'Email Address', className: 'email', type: 'email',
  },
  {
    input: 'Password',
    className: 'password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    type: 'password',
    title: 'Invalid Password',
  },
  {
    input: 'Confirm Password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    className: 'confirmPassword',
    type: 'password',
    title: 'Invalid Password',
  },
];
export const signupFirstNameLastNameInputs = [
  { input: 'First Name', className: 'firstName', type: 'text' },
  { input: 'Last Name', className: 'lastName', type: 'text' },
];
export const extraInfoData = {
  link1: '/signin',
  link2: '/forgot-password',
  displayText1: 'Don\'t have an account',
  displayText2: 'signin here',
  displayText3: 'Forgot password?',
};
export const socialData = {
  displayGoogleText: 'Signup with google',
  displayFacebookText: 'Signup with facebook',
};
