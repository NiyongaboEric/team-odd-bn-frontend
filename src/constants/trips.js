import React from 'react';

export const tableHeads = [
  { className: 'tripIdNav', attribute: 'Trip Id', key: 'tripId' },
  { className: 'tripTypeNav', attribute: 'Trip Type', key: 'tripType' },
  { className: 'createdAtNav', attribute: 'Created At', key: 'createdAt' },
  { className: 'status', attribute: 'Status', key: 'status' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];
export const navs = [
  { className: 'tableHeader', attribute: <h2>My Travel Requests</h2>, key: 'tableHeader' },
  { className: 'search', attribute: <input type="text" placeholder="search by date" />, key: 'search' },
];
