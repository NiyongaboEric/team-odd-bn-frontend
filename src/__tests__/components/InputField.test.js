import React from 'react';
import { shallow } from 'enzyme';
import InputField from '../../components/InputField';
import { signupInputs } from '../../constants/signUp';
import loginInputs from '../../constants/login';
describe('First React component test with Enzyme', () => {
  it('should call onChange prop for Login page', () => {
    expect(shallow(<InputField inputList={loginInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <InputField
        value=""
        placeholder=""
        inputList={[{ input: 'matched', className: 'matched' }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('input');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ matched: 'matched' });
  });

  it('should call onChange prop for signup', () => {
    expect(shallow(<InputField inputList={signupInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <InputField
        value=""
        placeholder=""
        inputList={[{ input: 'matched', className: 'matched' }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('input');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ matched: 'matched' });
  });
});
