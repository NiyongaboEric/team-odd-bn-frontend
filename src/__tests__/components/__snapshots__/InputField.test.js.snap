// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First React component test with Enzyme should call onChange prop 1`] = `ShallowWrapper {}`;
